app/scripts/metamask-controller.js

```diff
@@ -2780,19 +2787,16 @@ export default class MetamaskController extends EventEmitter {
       'PreferencesController:stateChange',
       previousValueComparator(async (prevState, currState) => {
         const { currentLocale } = currState;
-        const chainId = this.#getGlobalChainId();
 
-        await updateCurrentLocale(currentLocale);
+        const filtredChainIds = this.#getAllAddedNetworks().filter(
+          (networkId) =>
+            currState?.incomingTransactionsPreferences?.[networkId],
+        );
 
-        if (currState.incomingTransactionsPreferences?.[chainId]) {
-          this.txController.stopIncomingTransactionPolling();
+        await updateCurrentLocale(currentLocale);
 
-          this.txController.startIncomingTransactionPolling([
-            this.#getGlobalChainId(),
-          ]);
-        } else {
-          this.txController.stopIncomingTransactionPolling();
-        }
+        this.txController.stopIncomingTransactionPolling();
+        this.txController.startIncomingTransactionPolling(filtredChainIds);
```

Comment:
Nice!

It seems we do a lot of stop/starting? Is it worth refactoring this into something reusable?

```ts
#restartTransactionPoller() {
  const filteredChainIds = ..
  
  this.txController.stopPolling()
  
  if (filteredChainIds.length > 0) {
    this.txController.startPolling(filteredChainIds)
  }
}
```

Reply:
Open to leaving as is if we have different logic everywhere.